src/client/editor/models.js

```diff
@@ -16,44 +17,101 @@ function Model(name, sceneObject, projectData) {
 	this.concaveGeo = null; // For colliders. null - not generated, false - invalid
 	this.element = null;
 	this.prefabEntities = {};
+	this.childMeshes = [];
+	this.changed = false;
 
 	// Add to scene
 	modelsTab.group.add(this.sceneObject);
 	this.sceneObject.visible = false;
 
 	// Deep clone for editor
 	this.element = document.getElementById("modelTemplate").cloneNode(true);
+	this.element.classList.remove("itemTemplate");
 
 	// Add to model list
 	this.element.id = name;
-	this.element.getElementsByClassName("name")[0].innerHTML = name;
-	this.element.getElementsByClassName("name")[0].addEventListener("mousedown", function() {
+	this.element.getElementsByClassName("itemName")[0].innerHTML = name;
+	this.element.getElementsByClassName("showItem")[0].addEventListener("mousedown", function() {
 		modelsTab.select(name);
 	}, false);
 
+	// List model children
+	let childMeshes = _getChildMeshes(this.sceneObject);
+	for (let i = 0; i < childMeshes.length; i++) {
+		if (!this.projectData.childMeshes[i]) {
+			this.projectData.childMeshes[i] = {};
```

Comment:
I think in this case it's required to check the `projectData`. When adding a new model, it's impossible to know how many childMeshes there are. Unless you would prefer this section to be in the `modelsTab.loadModel()` function? That would give even more headaches, because `ChildMesh` relies on the parent `Model`...

Reply:
Sorry for the confusion, I meant how `this.projectData.childMeshes[i]` is set to an empty object in this case. The object would ultimately have a `material` property, but I don't think this is actually set anywhere.